include/alp/amf-based/vector.hpp

```diff
@@ -110,6 +119,152 @@ namespace alp {
 			typedef Matrix< T, structures::General, Density::Dense, View, ImfR, ImfC, backend > base_type;
 
 		private:
+			class VectorIterator: 
+				public std::iterator< std::random_access_iterator_tag, T > {
+				
+				friend class Vector< T, structures::General, Density::Dense, View, ImfR, ImfC, backend >;
```

Comment:
Does ALP Vector need to be a friend of the iterator?

Reply:
Yes to allow access to private constructors